src/languages/ilanguagesservice.h

```diff
@@ -35,14 +35,15 @@ class ILanguagesService : MODULE_EXPORT_INTERFACE
 public:
     virtual ~ILanguagesService() = default;
 
-    virtual ValCh<LanguagesHash> languages() const = 0;
+    virtual const LanguagesHash& languages() const = 0;
     virtual ValCh<Language> currentLanguage() const = 0;
 
-    virtual LanguageStatus::Status languageStatus(const QString& languageCode) const = 0;
+    virtual bool hasPlaceholderLanguage() const = 0;
+    virtual const Language& placeholderLanguage() const = 0;
 
-    virtual RetCh<LanguageProgress> install(const QString& languageCode) = 0;
-    virtual RetCh<LanguageProgress> update(const QString& languageCode) = 0;
-    virtual Ret uninstall(const QString& languageCode) = 0;
+    virtual LanguageProgressChannel update(const QString& languageCode) = 0;
+
+    virtual ValCh<bool> needRestartToApplyLanguageChange() const = 0;
```

Comment:
I would suggest not using ValCh. Use two separate methods here (one for notification and one for getting the actual value). We decided to drop this class in the future. For now, we should at least not use it in new parts of the code

Reply:
I will remove it, but actually I have started to like it, since it reduces the amount of members required to do a relatively simple thing. So, just for my understanding, what is wrong with it?